packages/shared_preferences/shared_preferences_tool/lib/src/shared_preferences_tool_eval.dart

```diff
@@ -116,10 +118,59 @@ class SharedPreferencesToolEval {
 }
 
 extension on EvalOnDartLibrary {
+  /// Evaluates the given [method] on the shared preferences instance.
+  ///
+  /// Returns the [InstanceRef] of the result.
+  /// The [isAlive] parameter is used to dispose the evaluation if the
+  /// caller is disposed.
+  ///
+  /// This method is actually a workaround for the asyncEval method, which is
+  /// not working for web targets.
   Future<InstanceRef> prefsEval(String method, Disposable? isAlive) async {
-    return (await asyncEval(
-      '(await SharedPreferences.getInstance()).$method',
+    // Create a empty list in memory to hold the shared preferences instance.
+    // It could've been anything that can handle values passed by reference.
+    final InstanceRef prefsHolderRef = await safeEval(
+      '[]',
+      isAlive: isAlive,
+    );
+
+    // Add the shared preferences instance to the list once the future completes
+    await eval(
```

Comment:
@kenzieschmoll this is the workaround that I've mentioned. Basically I'm getting the `SharedPreferences.getInstance()` value without calling the asyncEval. I took inspiration on the asyncEval implementation, but replaced the  `postEvent` call with an hack to save the value on a list (just to get the value by reference).

Let me know what you think, if you think this is too much of a hack I can revert the commits related to this change and we release without the web version for now.